pandas/core/window/rolling.py

```diff
@@ -437,37 +459,81 @@ def _apply(
                 results.append(values.copy())
                 continue
 
-            # if we have a string function name, wrap it
-            if isinstance(func, str):
-                cfunc = getattr(libwindow, func, None)
-                if cfunc is None:
-                    raise ValueError(
-                        "we do not support this function "
-                        "in libwindow.{func}".format(func=func)
-                    )
+            # Temporary path for our POC
+            if name == "mean":
```

Comment:
Can you please deduplicate the code is both inside the `if name == "mean"` and the else

Reply:
Deduplicated code around how `center` was checked and how `calc` was calculated.